013-large-sum/src/main.rs

```diff
@@ -0,0 +1,164 @@
+// Project Euler # 13
+// Work out the first ten digits of the sum of the following one-hundred 50-digit numbers.
+
+#[derive(Clone)]
+struct Total {
+    num: Vec<char>,
+}
+
+impl std::fmt::Display for Total {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", String::from_iter(self.num.clone().iter().rev()))
+    }
+}
+
+impl Total {
+    fn new() -> Total {
+        Total {
+            num: "0".chars().collect(),
+        }
+    }
+
+    fn add(&mut self, add_num: &str) {
+        let max_len = self.num.len().max(add_num.len());
+        let v1 = self
+            .num
+            .clone()
+            .into_iter()
+            .chain(std::iter::repeat('0'))
+            .take(max_len);
+
+        let v2 = add_num
+            .chars()
+            .rev()
+            .chain(std::iter::repeat('0'))
+            .take(max_len);
+
+        self.num = vec![];
+        let mut recip = 0;
+
+        for (a, b) in v1.zip(v2) {
+            let sum = a.to_digit(10).unwrap() + b.to_digit(10).unwrap() + recip;
+            self.num.push(char::from_digit(sum % 10, 10).unwrap());
+            recip = sum / 10;
+        }
+
+        while recip > 0 {
+            self.num.push(char::from_digit(recip % 10, 10).unwrap());
+            recip = recip / 10;
+        }
+    }
+}
+
+fn main() {
+    let mut total = Total::new();
+
+    for num_str in NUMS {
+        total.add(num_str);
+    }
+
+    println!("Total is {}", total);
+}
+
+const NUMS: &'static [&'static str] = &[
```

Comment:
this doesnt need to be static.  Constants have a default `'static` lifetime.

Reply:
Yup, still running fan after removing the `'static`. I think I added those while debugging compiler issues and just left them in, now realizing.